### PR TITLE
[FLINK-14560] [runtime] The value of taskmanager.memory.size in flink…

### PR DESCRIPTION
**Pull request by faaronzheng**
…-conf.yaml is set to zero will cause taskmanager not to work

## What is the purpose of the change

In order to fix the issue FLINK-14560

## Brief change log

ConfigurationParserUtils.java : The comparison with the default value is removed, and only the existence is judged

TaskManagerOptions.java : the default value of MANAGED_MEMORY_SIZE is removed

LocalStreamEnvironment.java ： Delete assignment statement of MANAGED_MEMORY_SIZE

## Verifying this change

This change is already covered by existing tests, such as (please describe tests).
TaskManagerHeapSizeCalculationJavaBashTest.compareHeapSizeShellScriptWithJava()
NettyShuffleEnvironmentConfigurationTest.calculateHeapSizeMB()
NetworkBufferCalculationTest.calculateNetworkBufFromHeapSize()
YarnConfigurationITCase.testFlinkContainerMemory()

## Does this pull request potentially affect one of the following parts:

Dependencies (does it add or upgrade a dependency): (no)
The public API, i.e., is any changed class annotated with @Public(Evolving): (no)
The serializers: (no)
The runtime per-record code paths (performance sensitive): ( no)
Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes)
The S3 file system connector: (no)

## Documentation

Does this pull request introduce a new feature? (no)
If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/util/ConfigurationParserUtils.java [hunk 1]
```diff
@@ -42,26 +43,9 @@
...
-				managedMemorySize = MemorySize.parse(
-					configuration.getString(TaskManagerOptions.MANAGED_MEMORY_SIZE), MEGA_BYTES).getMebiBytes();
-			} catch (IllegalArgumentException e) {
-				throw new IllegalConfigurationException("Could not read " + TaskManagerOptions.MANAGED_MEMORY_SIZE.key(), e);
-			}
-		} else {
-			managedMemorySize = Long.valueOf(managedMemorySizeDefaultVal);
-		}
-
-		checkConfigParameter(configuration.getString(
```
[the remark this answers is not in the document]
Comment (faaronzheng) [reply flink-runtime/src/main/java/org/apache/flink/runtime/util/ConfigurationParserUtils.java]: If this check is necessary, TaskExecutorResourceUtils.getManagedMemorySize() should also check it.

File: flink-runtime/src/main/java/org/apache/flink/runtime/util/ConfigurationParserUtils.java [hunk 2]
```diff
@@ -43,24 +44,16 @@
...
+		} catch (IllegalArgumentException e) {
+			throw new IllegalConfigurationException("Could not read " + TaskManagerOptions.MANAGED_MEMORY_SIZE.key(), e);
 		}
-
-		checkConfigParameter(configuration.getString(
-			TaskManagerOptions.MANAGED_MEMORY_SIZE).equals(TaskManagerOptions.MANAGED_MEMORY_SIZE.defaultValue()) || managedMemorySize > 0,
+		checkConfigParameter( managedMemorySize > 0,
 			managedMemorySize, TaskManagerOptions.MANAGED_MEMORY_SIZE.key(),
 			"MemoryManager needs at least one MB of memory. " +
 				"If you leave this config parameter empty, the system automatically pick a fraction of the available memory.");
```
Comment (tillrohrmann) [inline flink-runtime/src/main/java/org/apache/flink/runtime/util/ConfigurationParserUtils.java]: I think we also need to check whether the configuration contains `MANAGED_MEMORY_SIZE` as we have just removed the default value from the `ConfigOption`. Only if `MANAGED_MEMORY_SIZE` is defined we should parse the memory size and check that it is larger than 0. If not, then we should simply return `0`.

Comment (faaronzheng) [reply flink-runtime/src/main/java/org/apache/flink/runtime/util/ConfigurationParserUtils.java]: Thanks for your attention, MemorySize.parse() will check if the MANAGED_MEMORY_SIZE is null. Do we need to check it again?

**Comment by faaronzheng** [comment]
> ## CI report:
> * [0947500](https://github.com/apache/flink/commit/0947500a3ee71f62fd992e8b856f1f6c9d9d68cb) : FAILURE [Build](https://travis-ci.com/flink-ci/flink/builds/135081985)
> * [19aa6c8](https://github.com/apache/flink/commit/19aa6c886e3084b1a00351754c9e9eed39fd8019) : PENDING [Build](https://travis-ci.com/flink-ci/flink/builds/135644725)
> 
> Bot commands

@flinkbot run travis

**Comment by faaronzheng** [comment]
> ## CI report:
> * [0947500](https://github.com/apache/flink/commit/0947500a3ee71f62fd992e8b856f1f6c9d9d68cb) : FAILURE [Build](https://travis-ci.com/flink-ci/flink/builds/135081985)
> * [19aa6c8](https://github.com/apache/flink/commit/19aa6c886e3084b1a00351754c9e9eed39fd8019) : FAILURE [Build](https://travis-ci.com/flink-ci/flink/builds/135644725)
> * [974d399](https://github.com/apache/flink/commit/974d39956940cfd53114c9f8667b80235d5c0a99) : FAILURE [Build](https://travis-ci.com/flink-ci/flink/builds/135653995)
> 
> Bot commands

@flinkbot run travis

**Comment by faaronzheng** [comment]
> * @flinkbot run travis

@flinkbot run travis

**Comment by faaronzheng** [comment]
> ## CI report:
> * [0947500](https://github.com/apache/flink/commit/0947500a3ee71f62fd992e8b856f1f6c9d9d68cb) : FAILURE [Build](https://travis-ci.com/flink-ci/flink/builds/135081985)
> * [19aa6c8](https://github.com/apache/flink/commit/19aa6c886e3084b1a00351754c9e9eed39fd8019) : FAILURE [Build](https://travis-ci.com/flink-ci/flink/builds/135644725)
> * [974d399](https://github.com/apache/flink/commit/974d39956940cfd53114c9f8667b80235d5c0a99) : CANCELED [Build](https://travis-ci.com/flink-ci/flink/builds/135653995)
> * [c1bd674](https://github.com/apache/flink/commit/c1bd67424db724990265fa7e42629c0c94fe0591) : FAILURE [Build](https://travis-ci.com/flink-ci/flink/builds/135668832)
> 
> Bot commands
@flinkbot run travis

**Comment by faaronzheng** [comment]
> * @flinkbot run travis

@flinkbot run travis

**Comment by faaronzheng** [comment]
> ## CI report:
> * [0947500](https://github.com/apache/flink/commit/0947500a3ee71f62fd992e8b856f1f6c9d9d68cb) : FAILURE [Build](https://travis-ci.com/flink-ci/flink/builds/135081985)
> * [19aa6c8](https://github.com/apache/flink/commit/19aa6c886e3084b1a00351754c9e9eed39fd8019) : FAILURE [Build](https://travis-ci.com/flink-ci/flink/builds/135644725)
> * [974d399](https://github.com/apache/flink/commit/974d39956940cfd53114c9f8667b80235d5c0a99) : CANCELED [Build](https://travis-ci.com/flink-ci/flink/builds/135653995)
> * [c1bd674](https://github.com/apache/flink/commit/c1bd67424db724990265fa7e42629c0c94fe0591) : CANCELED [Build](https://travis-ci.com/flink-ci/flink/builds/135668832)
> * [8c36dff](https://github.com/apache/flink/commit/8c36dfff0b03d50e364bb0a64a1964f80ed09dcb) : FAILURE [Build](https://travis-ci.com/flink-ci/flink/builds/135805733)
> 
> Bot commands

@flinkbot run travis

**Pull request closed without merging**